proposals/apple-mdm/Fleet-MDM-Plan.md

```diff
@@ -0,0 +1,218 @@
+        munkitools["munkitools-*.pkg<br>(signed)<br>/mdm/apple/munki/pkg"]
+    end
+    subgraph MySQL
+        direction LR;
+        mdmAppleDB[(nanoMDM<br>Schemas)];
+        fleetDB[(fleet)];
+    end
+end
+
+ApplePush[https://api.push.apple.com];
+AppleDEP[https://mdmenrollment.apple.com];
+nanoDEP -- Apple MDM DEP API ----> AppleDEP;
+nanoMDM --> ApplePush;
+
+nanoDEP --> mdmAppleDB;
+nanoMDM --> mdmAppleDB;
+```
+
+## Plan for Dogfood-MVP
+
+### Setup
+
+See [Fleet MDM setup](./Fleet-MDM-Setup.md).
+
+### Authority
+
+Only global admins will be allowed to use MDM functionality.
+
+### MDM features
+
+#### Enrollments
+
+"Enrollments" hold some settings for devices that will be enrolled to MDM.
+The MDM "enrollments" will allow Fleet to automatically enroll devices to specific teams, which then allows for applying specific MDM settings (depending on the team).
+
+For Dogfood-MVP, Fleet will allow creating global enrollments only (team support will be added at a subsequent iteration).
+Users will be able to create the two following types of enrollments:
+- Global manual enrollment
+- Global DEP enrollment
+
+We'll have a new `apple_manual_enrollments` table with the following fields:
+- ID (to deduce an "Enroll URL")
+- Name
+- Configuration (JSON?)
+- Team (or global)
+- Authentication configuration (initially NULL, meaning not authenticated)
+
+We'll have a new `apple_dep_enrollments` table with the following fields:
+- ID
+- Name
+- Configuration (JSON?)
+- Team (or global)
+- Authentication configuration (initially NULL, meaning not authenticated)
+- Reference to `dep_names` table (will be NULL at first)
+
+Fleetctl commands (and APIs):
+- Create enrollments:
+	`fleetctl apple-mdm enrollments create --configuration=<TBD> --name=Foo`
+
+- List enrollments (the "global" manual enroll and the DEP enroll):
+	`fleetctl apple-mdm enrollments list`
+
+##### DEP enrollment setup
+
+These commands use the <ENROLLMENT_ID>, which is the `enrollments.ID` (obtained via `fleetctl apple-mdm enrollments list`).
+The token downloaded from https://business.apple.com basically links a "DEP enrollment" in Fleet (entry in `apple_dep_enrollments`) to what Apple calls an "MDM server" in https://business.apple.com.
+
+###### 1. `fleetctl apple-mdm enrollments dep set-token setup --enrollment=<ENROLLMENT_ID>`
+
+- Checks the `ENROLLMENT_ID` is a DEP enrollment.
+- Generates `fleet-mdm-apple-dep.pem` and `fleet-mdm-apple-dep.key`:
+	- Stores `fleet-mdm-apple-dep.pem` as a file.
+	- Keeps `fleet-mdm-apple-dep.key` in `~/.fleet/config` under <ENROLLMENT_ID>.
+
+###### 2. Upload PEM to Apple
+
+User uploads `fleet-mdm-apple-dep.pem` to https://business.apple.com, and downloads a `*.p7m` file. Let's call it `fleet-mdm-apple-dep-auth-token-encrypted.p7m`.
+
+###### 3. Finalize DEP setup
+
+`fleetctl apple-mdm enrollments dep set-token finalize --enrollment=<ENROLLMENT_ID> --encrypted-auth-token=fleet-mdm-apple-dep-auth-token-encrypted.p7m`
```

Comment:
Seems like this command/flow is more complex than needed, as I understand so far. 

Could it be `fleetctl apple-mdm dep token generate <ENROLLMENT_ID>` which generates the .key and .pem. Then `fletctl apple-mdm dep token finalize <ENROLLMENT_ID> <p7m_file> for the second step?

Reply:
> Seems like this command/flow is more complex than needed, as I understand so far.

What's the complexity? Using flag names? `--enrollment=` and `--encrypted-auth-token=`?